api/src/main/java/jakarta/data/page/impl/KeysetAwarePageRecord.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+package jakarta.data.page.impl;
+
+import jakarta.data.page.KeysetAwarePage;
+import jakarta.data.page.PageRequest;
+
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * Record type implementing {@link KeysetAwarePage}.
+ * This may be used to simplify implementation of a repository interface.
+ *
+ * @param content The page content, that is, the query results, in order
+ * @param cursors A list of {@link PageRequest.Cursor} instances for result,
+ *                in order
+ * @param totalElements The total number of elements across all pages that
+ *  *                   can be requested for the query
+ * @param pageRequest The {@link PageRequest page request} for which this
+ *                    slice was obtained
+ * @param nextPageRequest A {@link PageRequest page request} for the next
+ *                        page of results
+ * @param previousPageRequest A {@link PageRequest page request} for the
+ *                            previous page of results
+ * @param <T> The type of elements on the page
+ */
+public record KeysetAwarePageRecord<T>
+		(List<T> content, List<PageRequest.Cursor> cursors, long totalElements, PageRequest<T> pageRequest,
```

Comment:
Are you sure it is a good idea to precompute all of the cursors?  When I implemented this, I created each cursor only upon request, under the assumption that users will typically only want 1 or 2 cursors per page (usually for the next or previous page), and not a cursor for every result on the page.